src/main/scala/mesosphere/marathon/core/launchqueue/impl/OfferMatchStatisticsActor.scala

```diff
@@ -1,138 +0,0 @@
-package mesosphere.marathon
-package core.launchqueue.impl
-
-import akka.actor.{Actor, ActorRef, Props}
-import mesosphere.marathon.core.launcher.OfferMatchResult
-import mesosphere.mesos.NoOfferMatchReason
-
-import scala.collection.mutable
-import OfferMatchResult._
-import mesosphere.marathon.core.launchqueue.LaunchQueue.{QueuedInstanceInfoWithStatistics, QueuedInstanceInfo}
-import mesosphere.marathon.state.PathId
-
-/**
-  * The OfferMatchStatisticsActor is responsible for creating statistics for matched/unmatched offers based on RunSpecId.
-  * For every RunSpec this information is maintained:
-  * - offer matching statistics
-  * - the last offers that are unmatched per agent
-  */
-class OfferMatchStatisticsActor extends Actor {
```

Comment:
Ah, this explains the big diff. I thought this would stay and be queried. So you opted for removing this actor as well, @timcharper. Awesome! It seems like way more work than I'd have done but am happy that you wen the extra mile.

Reply:
Yeah, it's implemented as two different LiveFolds now, since there were two independent sets of statistics tracked by it.

LiveFold is something I introduced; it's just a fold that allows you to query the current result. When it comes to aggregating, few approaches are simpler than a fold.